tests/unit_tests/input_scan_tests.sh

```diff
@@ -31,4 +31,22 @@ test_scan_a_directory(){
     assertContains "$actual_message" "$expected_message2"
 }
 
+test_scan_valid_file_error(){
```

Comment:
this is a valid test, but it is actually testing the severity_mode input and so it belongs to the process_input_tests. 
As I discussed above, you need to set the severity_mode directly in the test case and then call process_input